protobuf/api/event.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+import "protobuf/definition/event.proto";
+
+package api;
+
+// This is the API to interact with the Events.
+//
+// This API is a [gRPC](https://grpc.io/) API.
+//
+// The source file of this API is hosted on [GitHub](https://github.com/mesg-foundation/engine/blob/master/protobuf/api/event.proto).
+service Event {
+  // Create creates event with data.
+  rpc Create(CreateEventRequest) returns (CreateEventResponse) {}
+
+  // Stream returns a stream of events that satisfy criteria
+  // specified in a request.
+  rpc Stream(StreamEventRequest) returns (stream definition.Event) {}
+}
+
+// StreamEventRequest defines request to retrieve a stream of executions.
+message StreamEventRequest {
+  // Filter contains filtering criteria.
+  message Filter {
+    // hash to filter events.
+    string hash = 1;
+
+    // instance's hash to filter events.
+    string instanceHash = 2;
+
+    // key is the key of the event.
+    string key = 3;
+  }
+
+  // Filter used to filter a stream of events.
+  Filter filter = 1;
+}
+
+// CreateEventRequest defines request for execution update.
+message CreateEventRequest {
```

Comment:
you could change the message `CreateEventRequest` to:
```
message CreateEventRequest {
  // instanceHash is hash of instance that can proceed an execution.
  string instanceHash = 1;

   // key is the key of the event.
  string key = 2;

   // data is the data for the event.
  string data = 3;
}
```
So there is not need to check for a not empty hash in `server/grpc/api/event.go`

Also I think it's more logic to have different messages to create the resource and to return it.
The same modification should be done (but in a future time) on ServiceX/Create. Reusing the definition.service causes more problem than it solves.

Reply:
```
// The request's data for the `Create` API.
message CreateServiceRequest {
  definition.Service definition = 1; // The service definition to use to create the Service.
}
```

We accept service in servcie api, so I wanted to be consistent. And I prefer to check if hash is empty rather then redefine all event fields (same for `CreateServiceRequest`). It's more clear that CreateEventRquest expects event as argument.

Reply:
The following issue is actually to change `CreateServiceRequest`: https://github.com/mesg-foundation/engine/issues/1111